napari/layers/utils/text_manager.py

```diff
@@ -253,25 +252,23 @@ def compute_text_coords(
         anchor_y : str
             The vispy text anchor for the y axis
         """
-        if order is None:
-            order = range(ndisplay)
         anchor_coords, anchor_x, anchor_y = get_text_anchors(
             view_data, ndisplay, self.anchor
         )
+        # The translation should either be a scalar or be as long as
+        # the dimensionality of the associated layer.
+        # We do not have direct knowledge of that dimensionality, but
+        # can infer enough information to get the translation coordinates
+        # that need to offset the anchor coordinates.
         ndim_coords = min(ndisplay, anchor_coords.shape[1])
-        # broadcast in case translation is just a scalar
-        if self.translation.size == 1:
-            translation = np.broadcast_to(self.translation, ndim_coords)
-        else:
-            translation = self.translation
-        # get order of displayed dimensions
-        displayed_ordered = list(
-            reorder_after_dim_reduction(order[-ndim_coords:])
```

Comment:
Since the order passed through should have the same length as the layer's dimensionality, which should have the same length as the translation (when it's not a scalar!), we want to use the values in `order` directly to index `translation` to ensure we're getting the right offsets for what is being displayed. I think...

Reply:
I *think* it makes sense... ^^' this is really hard to get an intuition about :/

Reply:
Yeah. This can at least as an extra bump to make this logic more clear throughout napari.